roles/schema_registry/tasks/rbac.yml

```diff
@@ -30,19 +28,88 @@
   until: sr_mds_result.status == 204
   retries: "{{ mds_retries }}"
   delay: 5
+  no_log: "{{mask_secrets|bool}}"
   loop: "{{schema_registry_additional_system_admins}}"
   when: not ansible_check_mode
 
-- name: Grant Schema Registry user the role SecurityAdmin on the Schema Registry cluster
+- name: Grant Schema Registry OAuth user the role SecurityAdmin on the Schema Registry cluster
+  uri:
+    url: "{{mds_bootstrap_server_urls.split(',')[0]}}/security/1.0/principals/User:{{schema_registry_oauth_user}}/roles/SecurityAdmin"
+    method: POST
+    validate_certs: false
+    headers:
+      Content-Type: application/json
+      Authorization: "Bearer {{ authorization_token }}"
+    body_format: json
+    body: >
+      {
+        "clusters": {
+          "kafka-cluster": "{{kafka_cluster_id}}",
+          "schema-registry-cluster": "{{schema_registry_final_properties['schema.registry.group.id']}}"
+        }
+      }
+    status_code: 204
+  register: sr_mds_result
+  until: sr_mds_result.status == 204
+  retries: "{{ mds_retries }}"
+  delay: 5
+  no_log: "{{mask_secrets|bool}}"
+  when:
+    - not ansible_check_mode
+    - oauth_enabled
```

Comment:
As far as I understand the changes between ladap and idp are - caller Token and the principal. In that case can we try to reuse the code and make it parameterised please.

Reply:
No here, the reason we have added 2 separate tasks is for the case of ldap_with_oauth, when the customer is setting up ldap_with_oauth as greenfield setup, both sr_ldap_user and sr_oauth_user would need these permissions

Reply:
updated the rbac.yml to take sr_user as parameter